### PR TITLE
Perform the sync to the upstream repo

### PR DESCRIPTION
**Pull request by adglkh**
* Refactor to follow the upstream to get server setup scripts.
 * Add spread tests.
 * Add three commands. [revoke-client, connect-server, help]
 * Support hooks to set env variables.

### CODE REVIEW

File: bin/LICENSE [hunk 1]
```diff
@@ -1,23 +0,0 @@
-The MIT License (MIT)
```
Comment (cmars) [inline bin/LICENSE]: Parts of easy-openvpn-pkg (scripts, etc) are a derivative work based on docker-openvpn, and we should comply with the MIT license -- which is easy, it just requires attribution by including this file's license content.

I would recommend renaming this file to `LICENSE.docker-openvpn`, and add a small note above this line that says "This project contains source code derived from docker-openvpn, which is licensed as follows:".

It should be fine to release our modifications and additions under GPLv3, copyright Canonical Ltd. Consider adding a main LICENSE or COPYING file with the GPLv3 (I'm assuming that's your intended license, based on the comment headers in files below).

File: snapcraft.yaml [hunk 2]
```diff
@@ -1,10 +1,13 @@
 name: easy-openvpn
-version: 2.3.10-1
+version: 2.3.10-4
 summary: An easy-to-manage OpenVPN deployment.
 description: |
     OpenVPN snap with management scripts that simplify PKI and client
     management for fast, cheap and disposable VPNs. Based on
     https://github.com/kylemanna/docker-openvpn.
+    It is an open source project distributed under MIT License.
```
Comment (cmars) [inline snapcraft.yaml]: With the license file included, we don't need to declare a license here at all, this section is typically used for a "product description" for snaps. Maybe just:
```
 description: |
    OpenVPN snap with management scripts that simplify PKI and client
    management for fast, cheap and disposable VPNs.
    See the project homepage for more details:
    https://code.launchpad.net/~snappy-hwe-team/snappy-hwe-snaps/+git/easy-openvpn
```

File: bin/LICENSE (hunk 1 above)
Comment (adglkh) [reply bin/LICENSE]: Yes, It really makes sense.
I'll add a main LICENSE file later on per your suggestion and fine-tune the current LICENSE file in the next commit.

File: snapcraft.yaml (hunk 2 above)
Comment (adglkh) [reply snapcraft.yaml]: With above modification, the description of this snap is gonna be very neat.
Thanks again.

**Pull request merged**